specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2020-04-01-preview/managedHsm.json

```diff
@@ -0,0 +1,634 @@
+        },
+        "softDeleteRetentionInDays": {
+          "type": "integer",
+          "format": "int32",
+          "default": 90,
+          "description": "softDelete data retention days. It accepts >=7 and <=90."
+        },
+        "enablePurgeProtection": {
+          "type": "boolean",
+          "description": "Property specifying whether protection against purge is enabled for this managed HSM pool. Setting this property to true activates protection against purge for this managed HSM pool and its content - only the Managed HSM service may initiate a hard, irrecoverable deletion. The setting is effective only if soft delete is also enabled. Enabling this functionality is irreversible - that is, the property does not accept false as its value."
+        },
+        "createMode": {
+          "type": "string",
+          "description": "The create mode to indicate whether the resource is being created or is being recovered from a deleted resource.",
+          "enum": [
+            "recover",
+            "default"
+          ],
+          "x-ms-enum": {
+            "name": "CreateMode",
+            "modelAsString": false,
+            "values": [
+              {
+                "value": "recover",
+                "description": "Recover the vault from a soft-deleted resource."
+              },
+              {
+                "value": "default",
+                "description": "Create a new vault.  This is the default option."
+              }
+            ]
+          }
+        },
+        "statusMessage": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource Status Message."
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Provisioning state.",
+          "enum": [
+            "Succeeded",
+            "Allocating",
+            "Provisioning",
+            "Failed",
+            "Deleting",
+            "Activated",
+            "SecurityDomainRestore",
+            "Restoring",
+            "Updating"
+          ],
+          "x-ms-enum": {
+            "name": "ProvisioningState",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "Succeeded",
+                "description": "The managed HSM Pool has been full provisioned."
+              },
+              {
+                "value": "Allocating",
+                "description": "Resources are currently being allocated for the managed HSM resource."
+              },
+              {
+                "value": "Provisioning",
+                "description": "The managed HSM Pool is currently being provisioned."
+              },
+              {
+                "value": "Failed",
+                "description": "Provisioning of the managed HSM Pool has failed."
+              },
+              {
+                "value": "Deleting",
```

Comment:
Is this same as Soft-Deleted state in Amit's spec?

Reply:
To map Amit's spec should we have these two states instead of Deleting:  Soft-Deleted, Purging states ?

Reply:
Since we dont have a Purge action specifically for the clients, I think we can keep it "Deleting" for now, becaue user action is called "Delete". When we implement soft-delete we will change it to "Purging" corresponding to a 'Purge' Action.

Reply:
Also in case of soft delete, there will be additional properties like deleteDate, purgeScheduleDate etc. which means this will be a different object from MHSM resource, like Deleted MHSM resource. So there will be no need of a soft-deleted state in that case.